pkg/build/builder/docker.go

```diff
@@ -65,7 +65,7 @@ func (d *DockerBuilder) Build() error {
 		return err
 	}
 	defer removeImage(d.dockerClient, imageTag(d.build))
-	if d.build.Parameters.Output.Registry != "" || d.authPresent {
+	if len(d.build.Parameters.Output.DockerImageReference) != 0 {
```

Comment:
This is backward compatible, now that i've fixed build being encoded into an API version.  The conversion sets DockerImageReference, ImageTag, and Registry.  So old code still gets the values it expects.

----- Original Message -----
> > @@ -65,7 +65,7 @@ func (d *DockerBuilder) Build() error {
> >  		return err
> >  	}
> >  	defer removeImage(d.dockerClient, imageTag(d.build))
> > -	if d.build.Parameters.Output.Registry != "" || d.authPresent {
> > +	if len(d.build.Parameters.Output.DockerImageReference) != 0 {
> 
> i'm guessing @mfojtik 's sti builder builder custom image will need to
> consume this change, and anyone else who's produced a custombuilder so far
> (RCM?)  So does this not fall into the API backwards compatibility
> requirements?
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/679/files#r23220447